From da32c9dee975fddce5ea6e8867f7972a3edbff54 Mon Sep 17 00:00:00 2001
Message-Id: <da32c9dee975fddce5ea6e8867f7972a3edbff54.1667754775.git.razvan.deaconescu@cs.pub.ro>
From: Razvan Deaconescu <razvan.deaconescu@cs.pub.ro>
Date: Sat, 5 Nov 2022 17:30:55 +0200
Subject: [PATCH 1/3] src/include: If libc is present, include endian.h

When using Musl, the definition of LITTLE_ENDIAN and BIG_ENDIAN in
include/lwip/arch.h causes an warning of double define. This is because
Musl defines LITT_ENDIAN and BIG_ENDIAN without macro guards, expecting
it would be the first to include them.

This commit includes the endian.h header file before the definition in
include/lwip/arch.h thus remove the warning. The inclusion is guarded by
a check of libc (#if CONFIG_HAVE_LIBC).

Signed-off-by: Razvan Deaconescu <razvan.deaconescu@cs.pub.ro>
---
 src/include/lwip/arch.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/include/lwip/arch.h b/src/include/lwip/arch.h
index 58dae33..7337401 100644
--- a/src/include/lwip/arch.h
+++ b/src/include/lwip/arch.h
@@ -37,6 +37,10 @@
 #ifndef LWIP_HDR_ARCH_H
 #define LWIP_HDR_ARCH_H
 
+#ifdef CONFIG_HAVE_LIBC
+#include <endian.h>
+#endif
+
 #ifndef LITTLE_ENDIAN
 #define LITTLE_ENDIAN 1234
 #endif
-- 
2.17.1

